crates/bevy_ecs/src/storage/mod.rs

```diff
@@ -9,10 +9,235 @@ pub use resource::*;
 pub use sparse_set::*;
 pub use table::*;
 
+use crate::{
+    archetype::{Archetype, Archetypes},
+    component::{ComponentId, ComponentTicks, Components, StorageType},
+    entity::{Entity, EntityLocation},
+};
+use bevy_ptr::{OwningPtr, Ptr};
+use std::{any::TypeId, cell::UnsafeCell};
+
 /// The raw data stores of a [World](crate::world::World)
 #[derive(Default)]
 pub struct Storages {
     pub sparse_sets: SparseSets,
     pub tables: Tables,
     pub resources: Resources,
 }
+
+impl Storages {
+    /// Get a raw pointer to a particular [`Component`](crate::component::Component) and its [`ComponentTicks`] identified by their [`TypeId`]
+    ///
+    /// # Safety
+    /// - `entity_location` must be within bounds of the given archetype and `entity` must exist inside
+    /// - `component_id` must be valid
+    /// - `Archetypes` and `Components` must come from the world this of this `Storages`
+    /// - the caller must ensure that no aliasing rules are violated
+    #[inline]
+    pub unsafe fn get_component_and_ticks_with_type(
```

Comment:
These were `pub(crate)` before, why are they now just `pub`?

Reply:
I'd err on the side of caution here. Unless we have a solid external use case, having more `pub`, particularly for items that mutably change the state of the `World`, is just asking for public facing unsoundness.